documentation/modules/configuring/con-config-mirrormaker2-high-volume-messages.adoc

```diff
@@ -6,26 +6,53 @@
 
-The flush pipeline for data replication is *source topic -> (Kafka Connect) source message queue -> producer buffer -> target topic*.
-An offset flush timeout period (`offset.flush.timeout.ms`) is the time to wait for the producer buffer (`producer.buffer.memory`) to flush and offset data to be committed.
-Try to avoid a situation where a large producer buffer and an insufficient offset flush timeout period causes a _failed to flush_ or _failed to commit offsets_ type of error.
+MirrorMaker 2.0 uses an internal producer client (`KafkaProducer`) to replicate data from the source to target Kafka cluster. 
+To control the streaming of high volumes of data, you can tune the following producer configuration for optimal throughput: 
 
-This type of error means that there are too many messages in the producer buffer, so they can't all be flushed before the offset flush timeout is reached.
+* Producer batch size 
+* Producer buffer size 
 
-If you are getting this type of error, try the following configuration changes:
+.Producer batch size
+In certain circumstances, you might want to increase the size of the message batches sent in a single produce request.
+You increase the batch size using the `batch.size` producer configuration to reduce the number of outstanding messages ready to be sent.
+A larger batch size reduces the number of requests and the backlog in the message queue.
+A produce request is sent to the target cluster when the batch size is reached.
+By increasing the batch size, produce requests are delayed and more messages are added to the batch and sent to brokers at the same time.  
+This can improve throughput when you have just a few topic partitions that handle large numbers of messages. 
 
-* Decreasing the default value in bytes of the `producer.buffer.memory`
-* Increasing the default value in milliseconds of the `offset.flush.timeout.ms`
+NOTE: To be able to increase the batch size, the `connector.client.config.override.policy` must be set to `All`. This is the default setting. The property enables connector client configurations.   
+
+.Producer buffer size
+The data replication pipeline is as follows:
+
+*source topic -> (Kafka Connect tasks) source message queue -> producer buffer -> target topic*.
+
+The producer sends messages in its buffer to topics in the target Kafka cluster.
+While this is happening, Kafka Connect tasks continue to poll topics in the source Kafka cluster to add messages to the source message queue.
 
-The changes should help to keep the underlying Kafka Connect queue of outstanding messages at a manageable size.
-You might need to adjust the values to have the desired effect.
+The size of the buffer is set using the `producer.buffer.memory` property. 
+The producer waits for a specified timeout period (`offset.flush.timeout.ms`) before the buffer is flushed. 
```

Comment:
Rather than "the producer", I would use "the task" as subject here.

This is not true after the recent fix, where the offsets commit happens asynchronously in another thread. Rather than be completely blocked, MM2 would make very slow progress in case of high volume.